posts/azure-seen-by-developers.md

```diff
@@ -0,0 +1,190 @@
+
+![Figure 15 : "Firewalls and virtual networks" view](https://github.com/aversaveau/blog-usa/blob/azureSeenByDevs/images/2021/03/azure_seen_by_devs_15.png)
+
+This view allows you to create as many rules as needed to allow single IP addresses or ranges of IP addresses to connect to the DB. Developers can create rules to allow IP address ranges from their workplace to access remote DBs from their own computer.
+
+## Storages
+
+### Overview
+
+![Figure 16 : "Overview of storage accounts" view](https://github.com/aversaveau/blog-usa/blob/azureSeenByDevs/images/2021/03/azure_seen_by_devs_16.png)
+
+This view is the home page of a storage account. It provides access to the different storage services such as containers, file shares, tables and queues (we will only develop containers and queues). It also provides a quick link to the _Azure Storage_ online documentation.
+
+### Documentation
+
+![Figure 17 : Azure storage online documentation](https://github.com/aversaveau/blog-usa/blob/azureSeenByDevs/images/2021/03/azure_seen_by_devs_17.png)
+
+To configure Azure Storage in your application, you can follow [this documentation](https://docs.microsoft.com/fr-fr/azure/storage/common/storage-introduction), which redirects to the web page shown in Figure 16.
+
+The left-hand menu also contains useful shortcuts such as _Storage explorer_, which allows you to browse the tree structure of storage resources, _Access keys_, which allows you to retrieve application connection information to the storage code, and _Shared access signature_, which allows you to generate read and/or write access tokens for storage resources.
+
+![Figure 18 : Access keys for the connection to the storage](https://github.com/aversaveau/blog-usa/blob/azureSeenByDevs/images/2021/03/azure_seen_by_devs_18.png)
+
+This view provides the two key elements for the connection of a web app to an Azure storage: the Account name and the Account key. These two pieces of information will be entered in the application's environment variables and used by the Azure library when connecting to the storage of the environment concerned.
+
+![Figure 19 : the Shared access signature to obtain rights on the storage](https://github.com/aversaveau/blog-usa/blob/azureSeenByDevs/images/2021/03/azure_seen_by_devs_19.png)
+
+This view allows to generate access tokens (which are in fact URIs) providing restricted access rights to the storage and limited in time in order to secure the data stored there. Without a signature, a web app will not be able to access the resources of its storage. As far as signature generation is concerned, it is possible to generate a URI with read and write rights on all the storage resources for an unlimited period of time, but be careful about security. It is also possible to provide a read-only access token for one hour. Everything depends on the need.
```

Comment:
This view allows you to generate access tokens (which are in fact, URIs) providing restricted access rights to the storage and limited in time in order to secure the data stored. Without a signature, a web app will not be able to access the resources of its storage. As far as signature generation is concerned, it is possible to generate a URI with read-and-write rights on all the storage resources for an unlimited period of time, but be careful about security. It is also possible to provide a read-only access token for one hour. Everything depends on the _need_.